desktop/src/gui/menu_bar.rs

```diff
@@ -43,45 +43,42 @@ impl MenuBar {
         }
     }
 
-    pub fn show(
+    pub fn consume_shortcuts(
         &mut self,
-        locale: &LanguageIdentifier,
         egui_ctx: &egui::Context,
         dialogs: &mut Dialogs,
         mut player: Option<&mut Player>,
     ) {
-        egui::TopBottomPanel::top("menu_bar").show(egui_ctx, |ui| {
-            // TODO(mike): Make some MenuItem struct with shortcut info to handle this more cleanly.
-            if ui.ctx().input_mut(|input| {
-                input.consume_shortcut(&Self::SHORTCUT_OPEN_ADVANCED)
-            }) {
-                dialogs.open_file_advanced();
-            }
-            if ui.ctx().input_mut(|input| {
-                input.consume_shortcut(&Self::SHORTCUT_OPEN)
-            }) {
-                self.open_file(ui);
-            }
-            if ui.ctx().input_mut(|input| {
-                input.consume_shortcut(&Self::SHORTCUT_QUIT)
-            }) {
-                self.request_exit(ui);
-            }
-            if ui.ctx().input_mut(|input| {
-                input.consume_shortcut(&Self::SHORTCUT_PAUSE)
-            }) {
-                if let Some(player) = &mut player {
-                    player.set_is_playing(!player.is_playing());
-                }
+        // TODO(mike): Make some MenuItem struct with shortcut info to handle this more cleanly.
+        if egui_ctx.input_mut(|input| input.consume_shortcut(&Self::SHORTCUT_OPEN_ADVANCED)) {
+            dialogs.open_file_advanced();
+        }
+        if egui_ctx.input_mut(|input| input.consume_shortcut(&Self::SHORTCUT_OPEN)) {
+            self.open_file();
+        }
+        if egui_ctx.input_mut(|input| input.consume_shortcut(&Self::SHORTCUT_QUIT)) {
+            self.request_exit();
+        }
+        if egui_ctx.input_mut(|input| input.consume_shortcut(&Self::SHORTCUT_PAUSE)) {
+            if let Some(player) = &mut player {
+                player.set_is_playing(!player.is_playing());
             }
-            if ui.ctx().input_mut(|input| {
-                input.consume_shortcut(&Self::SHORTCUT_FULLSCREEN)
-            }) {
-                if let Some(player) = &mut player {
-                    player.set_fullscreen(true);
-                }
+        }
+        if egui_ctx.input_mut(|input| input.consume_shortcut(&Self::SHORTCUT_FULLSCREEN)) {
+            if let Some(player) = &mut player {
+                player.set_fullscreen(true);
```

Comment:
If now this shortcut works in full screen, it would perhaps be better to toggle full screen here so that pressing F11 will exit full screen.

Reply:
Good idea, I'll wait for #17592 to get merged and then I will make this toggle instead.

Reply:
Changed.